gnat2goto/driver/tree_walk.adb

```diff
@@ -771,27 +771,20 @@ package body Tree_Walk is
             Pos_Number := Pos_Number + 1;
          end loop;
 
+         Append_Struct_Member (Array_Temp_Struct, Low_Expr);
+         Append_Struct_Member (Array_Temp_Struct, High_Expr);
+         Append_Struct_Member (Array_Temp_Struct, Malloc_Call_Expr);
+
          Append_Declare_And_Init (Symbol     => Literal_Temp,
                                   Value      => Array_Expr,
                                   Block      => Result_Block,
                                   Source_Loc => Source_Loc);
-         Append_Op (Result_Block,
-                    Make_Code_Decl (Symbol          => Array_Temp,
-                                    Source_Location => Source_Loc,
-                                    I_Type          => Result_Type));
-
-         Append_Op (Result_Block,
-                    Make_Code_Assign (Rhs             => Low_Expr,
-                                      Lhs             => First1_Mem_Expr,
-                                      Source_Location => Source_Loc));
-         Append_Op (Result_Block,
-                    Make_Code_Assign (Rhs             => High_Expr,
-                                      Lhs             => Last1_Mem_Expr,
-                                      Source_Location => Source_Loc));
-         Append_Op (Result_Block,
-                    Make_Code_Assign (Rhs             => Malloc_Call_Expr,
-                                      Lhs             => Data_Mem_Expr,
-                                      Source_Location => Source_Loc));
+         --  As long as symex is field-insensitive we need to initialise the
+         --  array structure with the information about allocated size.
+         Append_Declare_And_Init (Symbol     => Array_Temp,
+                                  Value      => Array_Temp_Struct,
+                                  Block      => Result_Block,
+                                  Source_Loc => Source_Loc);
```

Comment:
"create a temporary struct and assign it in one swoop to Array_Temp - so that Symex does not see the struct as having been changed after its creation and can therefore see it as constant - which means that the allocated size become visible"

Is this what you meant?
If so - I think you left this small bit out...  (which would however be rather vital for the comprehension of what you're doing here).  

Pass this information on to the next person please... or you're making life rather difficult for who follows you.

Reply:
I've added your message to the commit as well as the comment above this initialisation.